### ISSUE TITLE
Extract files from borg diff

### ISSUE THREAD

**Issue by knutov** [borgbackup/borg#3514]
I tried but I did not find this in documentation or in issues here.

I have repo with multiple snapshots.

There is `borg diff`, so I can build the list of changed files between some snapshots.

So, how to extract only changed files (files from  this diff)?

**Comment by ThomasWaldmann** [comment]
There is no such functionality. Maybe `borg mount -o versions ...` is interesting?

**Comment by knutov** [comment]
Yes, `borg mount -o versions ...` is possible to use but a lot of manual work required to use it in my case..

Is it hard to implement to extract only delta between snapshots? This can be also used as a partial workaround for #2872

My most frequest scenarios are backup websites from shared hosting and extract previous versions after website was hacked or when developer broke something in new version.

**Comment by ThomasWaldmann** [comment]
borg only has full snapshots (each archive contains all the files). so any "diff" / "delta" / "increment" has to be computed first by comparing 2 snapshots. not easy to integrate.

also, before we do that, I guess we'ld rather implement some "update capability" for extract, we already have a ticket for that.

**Comment by knutov** [comment]
Ok then, but still there is `borg diff` and `borg extract` with filters, is it possible to combine them? 

I mean something better then to prepare output of `borg diff` as filter and do the second call of `borg` with parameters, as there is a problem in case of big list of files and the size of arguments string (getconf ARG_MAX gives me 2621440 and I suppose there is some memory limits in additional becouse I had troubles with less number of parameters).

**Comment by eharris** [comment]
@knutov If you are reaching limits in the number of files you can specify on the command line for the `extract`, keep in mind that you can use the `borg extract --patterns-from` and put the files to extract in the specified file.

That being said, I would like a built in way of doing this rather than having to parse the output of `borg diff` and then supply a big list of files back to it.

**Comment by knutov** [comment]
@ThomasWaldmann I was not able to find via search, what is the ticket number for "update capability for extract"?

**Comment by ThomasWaldmann** [comment]
Guess I meant #1986.
